src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BranchSCMHead.java

```diff
@@ -36,32 +34,13 @@
 
     private static final long serialVersionUID = 1L;
 
-    /**
-     * Cache of the repository type. Will only be {@code null} for data loaded from pre-2.1.0 releases
-     *
-     * @since 2.1.0
-     */
-    // The repository type should be immutable for any SCMSource.
-    @CheckForNull
-    private final BitbucketRepositoryType repositoryType;
```

Comment:
Seems to be ok: https://stackoverflow.com/questions/16261383/delete-field-from-old-java-class-implementing-serializable/16261562